backend/windmill-worker/src/worker.rs

```diff
@@ -1124,6 +1155,29 @@ pub async fn run_worker(
     let job_completed_processor_is_done =
         Arc::new(AtomicBool::new(matches!(conn, Connection::Http(_))));
 
+    let is_agent_worker_allowed_to_enable_live_shell = || {
+        worker_name
+            .split("-")
+            .last()
+            .map(|suffix| {
+                check_if_ag_worker_has_specific_suffix(AgentWorkerSuffix::EnableLiveShell, suffix)
+            })
+            .unwrap_or(false)
+    };
+
+    
+    if worker_name.starts_with(WORKER_NAME_PREFIX) || is_agent_worker_allowed_to_enable_live_shell() {
+        tracing::debug!("Worker {} now listening for host tag", &worker_name);
+        start_interactive_worker_shell(
+            conn.clone(),
+            hostname.to_owned(),
+            worker_name.clone(),
+            killpill_rx.resubscribe(),
+            base_internal_url.to_owned(),
+            worker_dir.clone(),
+        );
+    }
```

Comment:
wrong place to start the worker shell, this will start as many worker shell as NUM_WORKERS, as discussed live it should have been above, in the function that start all the different workers